Libraries/Opc.Ua.Client/SessionAsync.cs

```diff
@@ -304,7 +304,7 @@ public partial class Session : SessionClientBatched, ISession
                 try
                 {
                     await base.CloseSessionAsync(null, false, ct).ConfigureAwait(false);
-                    CloseChannel();
+                    await base.CloseChannelAsync(ct).ConfigureAwait(false);
```

Comment:
Q: I always wonder what it means that a "close" is cancelled using cancellation token (ct).  In that case, are we sure that the socket is finally closed anyway or is the whole op abandoned?

Reply:
as discussed, channel should always be closed.